test/fragment.spec.js

```diff
@@ -196,46 +196,80 @@ describe('RelativeFragment', () => {
     });
   });
 
+  it('does greedy matching', () => {
+    const wrapper = mount(
+      <RelativeFragment forRoute='/play'>
+      <RelativeFragment forRoute='/c/:code'>
+        <p>fist</p>
+      </RelativeFragment>
+      <RelativeFragment forRoute='/c'>
+        <p>second</p>
+      </RelativeFragment>
+      <div>
+        <ul>
+          <li>
+          <RelativeFragment forRoute='/c/:code'>
+            <p>third</p>
+          </RelativeFragment>
+          </li>
+        </ul>
+      </div>
+      <RelativeFragment forRoute='/c/code'>
+        <p>fourth</p>
+      </RelativeFragment>
+      </RelativeFragment>,
+      fakeContext({
+        pathname: '/play/c/code',
+        route: '/play/c/:code'
+      })
+    );
+
+    expect(wrapper.containsMatchingElement(<p>fist</p>)).to.be.true;
+    expect(wrapper.containsMatchingElement(<p>second</p>)).to.be.false;
+    expect(wrapper.containsMatchingElement(<p>third</p>)).to.be.true;
+    expect(wrapper.containsMatchingElement(<p>fourth</p>)).to.be.false;
+  });
+
   describe('basic page-by-page routing', () => {
     // eslint-disable-next-line no-extra-parens
     const element = (
       <RelativeFragment forRoute='/'>
         <h1>App Title</h1>
         <RelativeFragment forRoute='/cheese'>
           <p>Cheese</p>
-          <RelativeFragment forRoute='/:type'>
-            <p>Cheese Type</p>
-          </RelativeFragment>
           <RelativeFragment forRoute='/gifs'>
             <p>Cheese Gifs</p>
           </RelativeFragment>
+          <RelativeFragment forRoute='/:type'>
+            <p>Cheese Type</p>
+          </RelativeFragment>
```

Comment:
All these fragments had to be reordered as it was matching with the general route first and not rendering the rest. So I guess that means its working 😄